lib/core/src/checkbox/checkbox.tsx

```diff
@@ -15,17 +15,42 @@ interface CheckboxStyle {
 
 export interface CheckboxProps {
 	// Controlled
+	/**
+	 * Control the current checkbox is selected or not
+	 */
 	checked?: boolean;
+	/**
+	 * Callback to set the checked value in controlled mode
+	 */
 	setChecked?: (checked: boolean) => void;
+	/**
+	 * Define whether the checkbox is indeterminate or not 
+	 */
 	indeterminate?: boolean;
 	// Uncontrolled
+	/**
+	 * The default checked value of checkbox
+	 */
 	defaultChecked?: boolean;
+	/**
+	 * Reference to the button element. Usually useful in uncontrolled mode.
+	 */
 	forwardedRef?: React.ForwardedRef<HTMLInputElement>;
 	// Body
+	/**
+	 * The label value for the checkbox
+	 */
 	children: React.ReactNode;
+	/**
+	 * Define whether the checkbox is disabled or not
+	 */
 	disabled?: boolean;
 }
 
+/**
+ * Checkboxes allow to select multiple options from an options list.
+ */
```

Comment:
Should not use the "Component X allows users to do Y" writing style. We should de-emphasize the component, and emphasize the users, even implicitly. Also, this description only focuses on the usage of checkboxes as a group, while in reality they are also often used standalone, like in "I agree with the EULA".

I suggest to cover more, with comparison to other similar components:

> Users use a Checkbox to select ("check") or deselect an option. Functional-wise, this is similar to Toggles (provide link to the Switcher > Toggle section). Meanwhile, a list of Checkboxes are used to select multiple options. This is different from a list of Radios (link to Radio > Group section) where only a single option can be selected at the same time.